backend/doc/migration.md

```diff
@@ -34,7 +34,397 @@ and activate the Tracim virtualenv:
+
+## Postgresql Support real Enum type
+
+Postgresql does support real enum whereas others database support enum as a "check" constraint.
+This means different code to support postgresql specificity in alembic.
+
+creating enum is not too complex:
+
+```python
+from alembic import op
+import sqlalchemy as sa
+
+def upgrade():
+    enum = sa.Enum("INTERNAL", "LDAP", name="authtype")
+    enum.create(op.get_bind(), checkfirst=False)
+    # ### commands auto generated by Alembic - please adjust! ###
+    with op.batch_alter_table("users") as batch_op:
+        batch_op.add_column(sa.Column("auth_type", enum, server_default="INTERNAL", nullable=False))
+    # ### end Alembic commands ###
+
+
+def downgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    with op.batch_alter_table("users") as batch_op:
+        batch_op.drop_column("auth_type")
+    sa.Enum(name="authtype").drop(op.get_bind(), checkfirst=False)
+    # ### end Alembic commands ###
+```
+
+see [ab7c7f4bcbc5_add_auth_type_enum_to_user.py]("../tracim_backend/migration/ab7c7f4bcbc5_add_auth_type_enum_to_user.py")
+
+But updating enum list a bit more tricky:
+
+```python
+from alembic import op
+import sqlalchemy as sa
+
+previous_auth_type_list = ["INTERNAL", "LDAP", "UNKNOWN"]
+new_elements = ["REMOTE"]
+new_auth_type_list = previous_auth_type_list + new_elements
+default_type = "INTERNAL"
+
+users = sa.Table("users", sa.MetaData(), sa.Column("auth_type"))
+
+
+def upgrade():
+    if op.get_context().dialect.name == "postgresql":
+        # INFO - G.M - 2018-11-27 - TO modify type in postgresq, we should
+        # create a new one set column type to this new one and remove old one
+        op.execute("ALTER TYPE authtype RENAME TO authtype_old;")
+        op.execute("ALTER TABLE users alter auth_type drop default;")
+        enum = sa.Enum(*new_auth_type_list, name="authtype")
+        enum.create(op.get_bind(), checkfirst=False)
+        with op.batch_alter_table("users") as batch_op:
+            batch_op.alter_column(
+                "auth_type",
+                type_=enum,
+                postgresql_using="auth_type::text::authtype",
+                server_default="INTERNAL",
+            )
+        op.execute("DROP TYPE authtype_old;")
+    else:
+        enum = sa.Enum(*new_auth_type_list, name="authtype")
+        enum.create(op.get_bind(), checkfirst=False)
+        with op.batch_alter_table("users") as batch_op:
+            batch_op.alter_column("auth_type", type_=enum)
+```
+
+
+see [182b9f7aa837_add_remote_value_to_auth_type_list.py]("../tracim_backend/migration/182b9f7aa837_add_remote_value_to_auth_type_list.py") for a complete working sample with downgrade.
+
+## Mysql require always type to be defined when altering column
+
+When updating a table, it seems not always require to set the type, for exemple if you need just
+to rename a column. Due to database specific behaviour, alembic require to always
```

Comment:
Due to differing behaviors across database engines, Alembic always requires setting the type.